python/polycap.pyx

```diff
@@ -527,7 +527,65 @@ cdef polycap_vector3 np2vector(np.ndarray[double, ndim=1] arr):
     return rv
 
 cdef tuple vector2tuple(polycap_vector3 vec):
-    return (vec.x, vec.y, vec.z)
+    return (vec.x, vec.y, vec.z) #TODO: make it return named tuple instead
+
+'''Class containing information about the simulated leak events such as position and direction, energy and transmission weights.
+'''
+cdef class Leak:
+    cdef polycap_leak *leak
+    cdef object coords
+    cdef object direction
+    cdef object elecv
+    cdef object weight
+    cdef int64_t n_refl
+
+    def __cinit__(self):
+        self.leak = NULL
+        self.coords = None
+        self.direction = None
+        self.elecv = None
+        self.weight = None
+        self.n_refl = 0
+
+    def __dealloc__(self):
+        '''Free a :ref:``Leak`` class and all associated data'''
+        if self.leak is not NULL:
+            polycap_leak_free(self.leak, self.size) #TODO: tricky discrepency between C and python Leak struct?
```

Comment:
Wat ge hier doet in C valt moeilijk elegant te vertalen naar Python.

Je zou beter voor uw leaks werken met een array van pointers ipv structs.
Daarna kan je itereren over alle pointer in de array en ze individueel free'en, en daarna de array zelf.

Maar dit zal redelijk wat herschrijven worden vrees ik? 😢

Reply:
Heb het nu op een andere manier aangepakt, al vrees ik dat ze niet echt koosjer is. Kzou het testen met valgrind, maar aangezien ik niet kan compileren door cython issues... :P